src/Pages/TodoPage.tsx

```diff
@@ -0,0 +1,255 @@
+import {
+  ChangeEvent,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useReducer,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import {
+  addTodo,
+  getTodos,
+  removeTodo,
+  updateTodo,
+} from '../api/todos';
+
+import { TodoFilter } from '../components/TodoFilter';
+import { TodoList } from '../components/TodoList';
+import { Todo } from '../types/Todo';
+import { filteredTodos } from '../utils/filter';
+import { ErrorMessage } from '../components/ErrorMessage';
+import { Error } from '../enums/Error';
+import { TodoForm } from '../components/TodoForm';
+import { reducer } from '../reducer';
+import { ReducerTempTodoType } from '../enums/ReducerTempTodoType';
+import { useLocalStorage } from '../hooks/useLocalStorage';
+import { INITIAL_STATE_TEMPTODO } from '../constants/initial_state_newTodo';
+import { Context } from '../context';
+
+export const TodoPage: React.FC = () => {
+  const {
+    user,
+    currentError,
+    setCurrentError,
+  } = useContext(Context);
+
+  const [searchParams] = useSearchParams();
+  const filter = searchParams.get('filter') || '';
+
+  const navigate = useNavigate();
+
+  const [userName, setUserName] = useLocalStorage<string>('userName', '');
+
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+  const [tempTodo, dispatch] = useReducer(reducer, INITIAL_STATE_TEMPTODO);
+
+  const [processingIds, setProcessingIds] = useState<number[]>([]);
+  const [isCreated, setIsCreated] = useState(false);
+
+  const [isArrow, setIsArrow] = useState(false);
+
+  const isProcessing = useCallback((id: number) => {
+    return !!processingIds.includes(id) || id === 0;
+  }, [processingIds]);
+
+  const onShowArrow = useMemo(() => {
+    return setIsArrow(todos.every((todo: Todo) => todo.completed));
+  }, [todos]);
+
+  const visibleTodos = useMemo(() => {
+    return filteredTodos(todos, filter);
+  }, [todos, filter]);
+
+  const getTodosFromServer = async () => {
+    if (!user) {
```

Comment:
you already check this in useEffect, so, do you need this ?

Reply:
![image](https://user-images.githubusercontent.com/116097498/224490099-e8966bcf-9306-4038-aa6e-224a273b6199.png)
TypeScript

Reply:
Than don't check in useEffect)